server/boot/user.js

```diff
@@ -631,4 +643,61 @@ module.exports = function(app) {
       return res.render('account/forgot');
     });
   }
+
+  function getReportUserProfile(req, res) {
+    const username = req.params.username.toLowerCase();
+    const { user } = req;
+    if (!user.email || !user.emailVerified) {
```

Comment:
This should be a middleware and both the get request and the api should block those that have not been authenticated.